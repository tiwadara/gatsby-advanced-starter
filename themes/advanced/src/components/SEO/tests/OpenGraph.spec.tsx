import OpenGraphTags from "../OpenGraph";
import {
  tagListHasEmptyValues,
  sampleSeoData,
  tagListHasUniqueKeys,
} from "./TestUtils";

describe("seo module OpenGraphTags", () => {
  it("generates correct tags on website pages", () => {
    expect.assertions(1);

    const generatedTags = OpenGraphTags({
      seoData: sampleSeoData.seoWebsite,
      websiteData: sampleSeoData.website,
      userData: sampleSeoData.user,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags on article pages", () => {
    expect.assertions(1);

    const generatedTags = OpenGraphTags({
      seoData: sampleSeoData.seoArticle,
      websiteData: sampleSeoData.website,
      userData: sampleSeoData.user,
      postData: sampleSeoData.post,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags when missing description", () => {
    expect.assertions(1);

    const generatedTags = OpenGraphTags({
      seoData: { ...sampleSeoData.seoArticle, description: undefined },
      websiteData: sampleSeoData.website,
      userData: sampleSeoData.user,
      postData: sampleSeoData.post,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags when missing UserData", () => {
    expect.assertions(1);

    const generatedTags = OpenGraphTags({
      seoData: sampleSeoData.seoArticle,
      websiteData: sampleSeoData.website,
      userData: undefined,
      postData: sampleSeoData.post,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("doesn't generate tags when missing SEO imageUrl", () => {
    expect.assertions(1);

    const generatedTags = OpenGraphTags({
      seoData: { ...sampleSeoData.seoArticle, imageUrl: undefined },
      websiteData: sampleSeoData.website,
      userData: sampleSeoData.user,
      postData: sampleSeoData.post,
    });

    expect(generatedTags).toStrictEqual([]);
  });

  it("doesn't generate empty tags", () => {
    expect.assertions(5);

    // Test for website pages
    const generatedWebsiteTags = OpenGraphTags({
      seoData: sampleSeoData.seoWebsite,
      websiteData: sampleSeoData.website,
      userData: sampleSeoData.user,
    });

    expect(tagListHasEmptyValues(generatedWebsiteTags)).toBe(false);

    // Test for article pages
    const generatedArticleTags = OpenGraphTags({
      seoData: sampleSeoData.seoArticle,
      websiteData: sampleSeoData.website,
      userData: sampleSeoData.user,
      postData: sampleSeoData.post,
    });

    expect(tagListHasEmptyValues(generatedArticleTags)).toBe(false);

    // Test for missing fbAppId case
    const generatedArticleTagsWithoutFbAppId = OpenGraphTags({
      seoData: sampleSeoData.seoArticle,
      websiteData: { ...sampleSeoData.website, fbAppId: undefined },
      userData: sampleSeoData.user,
      postData: sampleSeoData.post,
    });

    expect(tagListHasEmptyValues(generatedArticleTagsWithoutFbAppId)).toBe(
      false
    );

    // Test for missing description case
    const generatedArticleTagsWithoutDescription = OpenGraphTags({
      seoData: { ...sampleSeoData.seoArticle, description: undefined },
      websiteData: sampleSeoData.website,
      userData: sampleSeoData.user,
      postData: sampleSeoData.post,
    });

    expect(tagListHasEmptyValues(generatedArticleTagsWithoutDescription)).toBe(
      false
    );

    // Test for missing UserData case
    const generatedTagsWithoutUserData = OpenGraphTags({
      seoData: sampleSeoData.seoArticle,
      websiteData: sampleSeoData.website,
      userData: undefined,
      postData: sampleSeoData.post,
    });

    expect(tagListHasEmptyValues(generatedTagsWithoutUserData)).toBe(false);
  });

  it("generates unique keys", () => {
    expect.assertions(2);

    // Test for website pages
    const generatedWebsiteTags = OpenGraphTags({
      seoData: sampleSeoData.seoWebsite,
      websiteData: sampleSeoData.website,
      userData: sampleSeoData.user,
    });

    expect(tagListHasUniqueKeys(generatedWebsiteTags)).toBe(true);

    // Test for article pages
    const generatedArticleTags = OpenGraphTags({
      seoData: sampleSeoData.seoArticle,
      websiteData: sampleSeoData.website,
      userData: sampleSeoData.user,
      postData: sampleSeoData.post,
    });

    expect(tagListHasUniqueKeys(generatedArticleTags)).toBe(true);
  });
});
