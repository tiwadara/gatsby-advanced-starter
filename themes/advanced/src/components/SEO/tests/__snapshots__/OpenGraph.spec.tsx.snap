// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seo module OpenGraphTags generates correct tags on article pages 1`] = `
Array [
  <meta
    content="Post title"
    property="og:title"
  />,
  <meta
    content="article"
    property="og:type"
  />,
  <meta
    content="/posts/url"
    property="og:url"
  />,
  <meta
    content="/logos/logo-1024.png"
    property="og:image"
  />,
  <meta
    content="Post cover image alt text."
    property="og:image:alt"
  />,
  <meta
    content="Gatsby Advanced Starter"
    property="og:site_name"
  />,
  <meta
    content="Post description."
    property="og:description"
  />,
  <meta
    content="1825356251115265"
    property="fb:app_id"
  />,
  <meta
    content="2021-12-17T00:00:00.000Z"
    property="article:published_time"
  />,
  <meta
    content="2021-12-20T00:00:00.000Z"
    property="article:modified_time"
  />,
  <meta
    content="http://examples.opengraphprotocol.us/profile.html"
    property="article:author"
  />,
  <meta
    content="First"
    property="profile:first_name"
  />,
  <meta
    content="Last"
    property="profile:last_name"
  />,
  <meta
    content="AdvancedUser"
    property="profile:username"
  />,
  <meta
    content="Tech"
    property="article:section"
  />,
  <meta
    content="Tag1"
    property="article:tag"
  />,
  <meta
    content="Tag2"
    property="article:tag"
  />,
]
`;

exports[`seo module OpenGraphTags generates correct tags on website pages 1`] = `
Array [
  <meta
    content="Website title"
    property="og:title"
  />,
  <meta
    content="website"
    property="og:type"
  />,
  <meta
    content="/url"
    property="og:url"
  />,
  <meta
    content="/logos/logo-1024.png"
    property="og:image"
  />,
  <meta
    content="Website image alt description."
    property="og:image:alt"
  />,
  <meta
    content="Gatsby Advanced Starter"
    property="og:site_name"
  />,
  <meta
    content="Website description."
    property="og:description"
  />,
  <meta
    content="1825356251115265"
    property="fb:app_id"
  />,
]
`;

exports[`seo module OpenGraphTags generates correct tags when missing UserData 1`] = `
Array [
  <meta
    content="Post title"
    property="og:title"
  />,
  <meta
    content="article"
    property="og:type"
  />,
  <meta
    content="/posts/url"
    property="og:url"
  />,
  <meta
    content="/logos/logo-1024.png"
    property="og:image"
  />,
  <meta
    content="Post cover image alt text."
    property="og:image:alt"
  />,
  <meta
    content="Gatsby Advanced Starter"
    property="og:site_name"
  />,
  <meta
    content="Post description."
    property="og:description"
  />,
  <meta
    content="1825356251115265"
    property="fb:app_id"
  />,
  <meta
    content="2021-12-17T00:00:00.000Z"
    property="article:published_time"
  />,
  <meta
    content="2021-12-20T00:00:00.000Z"
    property="article:modified_time"
  />,
  <meta
    content="http://examples.opengraphprotocol.us/profile.html"
    property="article:author"
  />,
  <meta
    content="Tech"
    property="article:section"
  />,
  <meta
    content="Tag1"
    property="article:tag"
  />,
  <meta
    content="Tag2"
    property="article:tag"
  />,
]
`;

exports[`seo module OpenGraphTags generates correct tags when missing description 1`] = `
Array [
  <meta
    content="Post title"
    property="og:title"
  />,
  <meta
    content="article"
    property="og:type"
  />,
  <meta
    content="/posts/url"
    property="og:url"
  />,
  <meta
    content="/logos/logo-1024.png"
    property="og:image"
  />,
  <meta
    content="Post cover image alt text."
    property="og:image:alt"
  />,
  <meta
    content="Gatsby Advanced Starter"
    property="og:site_name"
  />,
  <meta
    content="1825356251115265"
    property="fb:app_id"
  />,
  <meta
    content="2021-12-17T00:00:00.000Z"
    property="article:published_time"
  />,
  <meta
    content="2021-12-20T00:00:00.000Z"
    property="article:modified_time"
  />,
  <meta
    content="http://examples.opengraphprotocol.us/profile.html"
    property="article:author"
  />,
  <meta
    content="First"
    property="profile:first_name"
  />,
  <meta
    content="Last"
    property="profile:last_name"
  />,
  <meta
    content="AdvancedUser"
    property="profile:username"
  />,
  <meta
    content="Tech"
    property="article:section"
  />,
  <meta
    content="Tag1"
    property="article:tag"
  />,
  <meta
    content="Tag2"
    property="article:tag"
  />,
]
`;
